### ISSUE TITLE
ControlPath uses feature not present in CentOS7's version of OpenSSH

### ISSUE THREAD

**Issue by ragingpastry** [metacloud/molecule#862]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible --version
ansible 2.3.1.0

molecule --version
molecule, version 2.0.0.0rc6
```

- Molecule installation method: pip
- Ansible installation method:pip

# Desired Behaviour
 Under CentOS7, when running Molecule test with the vagrant driver, Molecule will generate ansible_ssh_common_args in molecule/default/.molecule/ansible_inventory.yml. Molecule should generate usable arguments based on the hosts OpenSSH version

# Actual Behaviour (Bug report only)

Molecule instead is hardcoded to use the following ssh arguments:

molecule/lib/python2.7/site-packages/molecule/driver/base.py
```
154     def _get_ssh_connection_options(self):
155         return [
156             '-o UserKnownHostsFile=/dev/null',
157             '-o ControlMaster=auto',
158             '-o ControlPersist=60s',
159             '-o IdentitiesOnly=yes',
160             '-o StrictHostKeyChecking=no',
161             '-o ControlPath=~/.ansible/cp/%C',
```

`ControlPath=~/.ansible/cp/%C` is not valid in openssh-6.6. Instead we have to use `'ControlPath=~/.ansible/cp/%r@%h-%p'`

It would be awesome if we could have molecule detect the host OS version / OpenSSH version and use the correct arguments for that version. I could try to hack something up if you want.

**Comment by ragingpastry** [comment]
Would it be better to replace %C with %r@%h:%p? After taking another look at the code it might be a bit tricky to use os detection

**Comment by retr0h** [comment]
@ragingpastry implemented in 31d68aeb5008528ec493dec783b726aead894862.

Update your `molecule.yml` with the ssh config options you desire.  Molecule does not merge lists, so you need to provide everything when overriding `ssh_connection_options`.

```
driver:
  name: ec2
  ssh_connection_options:
    - -o UserKnownHostsFile=/dev/null',
    - -o ControlMaster=auto',
    - -o ControlPersist=60s',
    - -o IdentitiesOnly=yes',
    - -o StrictHostKeyChecking=no',
    - -o ControlPath=~/.ansible/cp/%r@%h-%p
```
